subprojects/core/src/main/java/org/gradle/internal/classpath/CopyingClasspathFileTransformer.java

```diff
@@ -39,13 +38,9 @@ public File transform(File source, CompleteFileSystemLocationSnapshot sourceSnap
             // Directories are ok to use outside the cache
             return source;
         }
-        for (AdditiveCache cache : additiveCaches) {
-            for (File root : cache.getAdditiveCacheRoots()) {
-                if (source.toPath().startsWith(root.toPath())) {
-                    // It's not expected that the file will be deleted, so assume it is ok to use outside the cache
-                    return source;
-                }
-            }
+        if (globalCacheLocations.isInsideGlobalCache(source.getAbsolutePath())) {
+            // It's not expected that the file will be deleted, so assume it is ok to use outside the cache
```

Comment:
When the code was talking about _additive(-only)_ caches, it made sense to derive this conclusion. Now it talks about global caches, and the conclusion doesn't seem logical anymore. I wonder how this could be clarified. Maybe just state that global caches are additive-only?

Reply:
I changed the comment.